Adding do your own filter

Frenchris opened the pull request.

kigiri commented on exercise/do-your-own-filter.js:
```diff
@@ -0,0 +1,134 @@
+const check7 = (el) => el % 2 === 0 && el % 5 === 0
+const check8 = (el) => el % 2 === 0 && el % 10 === 0
+const check9 = (el) => el % 3 === 0 && el % 5 === 0
+const check10 = (el) => el % 5 === 0 && el % 10 === 0
```
modulo again ? Isn't that a waste of opportunity to show of good use cases of each of those wonderfull functions ?

kigiri commented on tests/filter/pass_afonso.js:
```diff
@@ -0,0 +1,4 @@
+let arr2 = arr1.filter((x) => x % 2 === 0)
+let arr3 = arr1.filter((x) => x % 3 === 0)
+let arr4 = arr1.filter((x) => (x % 3 === 0) & (x % 2 === 0))
```
this one too

kigiri commented on tests/filter/pass_frenchris.js:
```diff
@@ -0,0 +1,7 @@
+let arr2 = arr1.filter((el) => el % 2 === 0)
+
```
and also this one

kigiri commented on exercise/filter.js:
```diff
@@ -0,0 +1,112 @@
+
+Create an array named 'arrRandom2' that contains the values of 'arrRandom1'
+that:
+-are divisible by 2 and 3.
```
this file seem to have been added by mistake

Frenchris committed:
module updated for "do your own filter" exercise to run

Frenchris committed:
updated subject for "do your own filter" exercise

Frenchris committed:
updated solution for "do your own filter" exercise (frenchris solution only working one)

Frenchris committed:
1) change of name of `filter` to `using-filter`
2) change of name of `do your own filter` to `filter`
3) updated  `filter` (old `do your own filter`) with an extra test with the use of object
4) added `Object.getPrototypeOf([]).proto = ' [avoid for..in] '` line to avoid the use of for...in
5) updated frenchris solution of map to comply with the [avoid for..in]

kigiri committed:
call-me-maybe/filter: fixes + solutions
